gym_sts/envs/base.py

```diff
@@ -338,32 +339,24 @@ def step(self, action_id: int) -> Tuple[dict, float, bool, dict]:
             # Maybe check that the new obs is the same as the old one, modulo
             # the error field?
             obs = prev_obs
-            valid_actions = ActionValidators.valid_actions(obs)  # use cached?
         else:
-            valid_actions = ActionValidators.valid_actions(obs)
             success = False
             for _ in range(10):
-                if len(valid_actions) == 0:
+                if len(obs.valid_actions) == 0:
                     # this can indicate instability
                     time.sleep(1)
                     obs = self.observe()
-                    valid_actions = ActionValidators.valid_actions(obs)
                 else:
                     success = True
                     break
             if not success:
                 raise StSError("No valid actions.")
 
-            reward = observation_value(obs) - observation_value(prev_obs)
+            reward = obs_value(obs) - obs_value(prev_obs)
```

Comment:
Can the old `observation_value()` function be deleted?

Reply:
Done